tools/squidclient/Transport.cc

```diff
@@ -249,8 +249,8 @@ Transport::Write(const void *buf, size_t len)
 #endif
     } else {
 
-#if _SQUID_WINDOWS_
-        return send(conn, buf, len, 0);
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+        return send(conn, static_cast<const char *>(buf), len, 0);
 #else
         alarm(Config.ioTimeout);
         return write(conn, buf, len);
```

Comment:
I am concerned about us adding MinGW-specific hacks to high-level code. For example, there are other send(2) calls in high-level Squid code. Some do not use `char *` buffers AFAICT. There ought to be write(2) calls as well. If we end up adding similar MinGW wrappers to many of them, we would be making our code worse overall, even though each specific change may look "small".

Similarly, AFAICT, this change effectively disables squidclient timeout feature on MinGW. What criteria do we use to decide which features are essential for MinGW and which can be easily dropped on MinGW? For example, perhaps the squidclient feature _itself_ can be dropped on MinGW? There ought to be other HTTP clients supported in that environment. I do not know the criteria you use. Can you share them?

If we decide that we must support squidclient on MinGW, then I see two possible next steps:

A. Investigate what it would take for MinGW builds to use the ordinary write() code path instead of the exceptional send() path. If MinGW can use that ordinary code path without heroic efforts, great! We get fewer exceptional handling in high-level Squid code now and can solve the send() argument type problem later.

B. If using write(2) is not feasible, provide a send(2) portability wrapper that can be used without casting and `#ifs` throughout Squid high-level code. AFAICT, old Windows code attempted to do that in compat/os/mswindows.h (but we may need a different solution for MinGW).